File: BOR1/frontend/src/types/permit.ts
export interface PermitRow {
  id: string;
  emissao: string;
  jobsite: string;
  lot_address: string;
  situacao: string;
  solicitacao: string;
  aplicacao: string;
  observacao: string;
  arquivo: string;
  model: string;
  descricao?: string;
  vencimento?: string;
  status?: string;
}

export interface PermitFilters {
  selectedYear: string;
  selectedMonth: string;
  selectedModel: string[];
  selectedSituation: string[];
  selectedJobsite: string[];
}

export interface PermitMetrics {
  totalPermits: number;
  pendingPermits: number;
  approvedPermits: number;
  rejectedPermits: number;
} 

File: BOR2/apps/api/internal/domain/permit.go
package domain

import "time"

// PermitRow maps to the permit_control table.
type PermitRow struct {
	ID          string     `json:"id"`
	Model       string     `json:"model"`
	Jobsite     string     `json:"jobsite"`
	LotAddress  string     `json:"lotAddress"`
	Situacao    string     `json:"situacao"`
	Solicitacao *time.Time `json:"solicitacao,omitempty"`
	Aplicacao   *time.Time `json:"aplicacao,omitempty"`
	Emissao     *time.Time `json:"emissao,omitempty"`
	Observacao  string     `json:"observacao"`
	Arquivo     string     `json:"arquivo"`
	CreatedAt   time.Time  `json:"createdAt"`
}

type PermitRowFilters struct {
	Jobsite  string
	Situacao string
}
